magento/magento2
# Compare admin domain from the start, not as a substring

### Description (*)


When custom admin URL is used, the current algorithm wrongly detects custom admin URL, when its in a form of subdomain (for example admin.domain.com)

### Related Pull Requests

None

### Fixed Issues (if relevant)

1. Fixes magento/magento2#37663

### Manual testing scenarios (*)

Install Magento and make sure it is working URL should be m2.domain.local

1. Enable and set the admin path to "/admin123" (in env.php), and make sure admin UI logs in after changing the admin path
2. Create a frontend (store_view) on the domain "m2.domain.local"
3. Set the custom admin URL to "admin.m2.domain.local". Also, make sure u r updating the virtual host and host files accordingly. Restart the httpd server
4. Access using a custom URL, ur application should work


### Questions or comments

None

### Contribution checklist (*)
